CMakeLists.txt

```diff
@@ -602,7 +602,7 @@ if(UNIX)
 endif()
 
 if(ENABLE_SDL)
-  dolphin_find_optional_system_library(SDL2 Externals/SDL 2.30.9)
+  dolphin_find_optional_system_library(SDL3 Externals/SDL 3.2.0)
```

Comment:
I suspect that we might need to enable `OBJC` on the top-level CMakeLists before including the SDL according to: https://gitlab.kitware.com/cmake/cmake/-/issues/24104#note_1269137

Hopefully, that will fix it, otherwise I'll be out of ideas.